File: coding/exponetiation.rb
def exponent(base, power)
    i = 0
    sum = 1
    return sum if power == 0

    if power < 1
        while i < power
        sum *= base
        i -= 1
        end

        return 1.0/sum
    else
        while i < power
            sum *= base
            i += 1
        end

        return sum
    end
end

File: coding/exponetiation.js
function exponent(base, power) {
    let product = 1;

    if (power === 0) return product;

    if (power < 0) {
        for (let i = 0; i > power; i--) {
            product *= base;
        }

        return 1/product;
    } else {
        for (let i = 0; i < power; i++) {
            product *= base;
        }

        return product;
    }
}
